content-src/asrouter/templates/Trailhead/Trailhead.jsx

```diff
@@ -15,26 +16,83 @@ const FLUENT_FILES = [
 export class Trailhead extends React.PureComponent {
   constructor(props) {
     super(props);
-    this.state = {isModalOpen: true};
     this.closeModal = this.closeModal.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onInputInvalid = this.onInputInvalid.bind(this);
+
+    this.state = {
+      emailInput: "",
+      isModalOpen: true,
+      flowId: "",
+      flowBeginTime: 0,
+    };
+    this.didFetch = false;
   }
 
-  componentWillMount() {
+  async componentWillMount() {
```

Comment:
is it ok to make this async? Is the rendering going to be delayed until the fetch happens below?

Reply:
I originally had it in `componentDidMount` but you aren't supposed to call `setState` from there.